wgsl/index.bs

```diff
@@ -3682,15 +3682,18 @@ function_decl
   : decoration_list* function_header body_statement
 
 function_type_decl
-  : type_decl
+  : decoration_list* type_decl
   | VOID
 
 function_header
   : FN IDENT PAREN_LEFT param_list PAREN_RIGHT ARROW function_type_decl
 
 param_list
   :
-  | (variable_ident_decl COMMA)* variable_ident_decl
+  | (param COMMA)* param
+
+param
+  : decoration_list* variable_ident_decl
```

Comment:
same here, perhaps `decoration_list?` is a better fit

Reply:
decoration_list* is the pattern everywhere else in the grammar.